math/src/fft/gpu/cuda/ops.rs

```diff
@@ -27,10 +27,11 @@ where
 
     let order = input.len().trailing_zeros();
     for stage in 0..order {
-        let group_count = 1 << stage;
-        let group_size = input.len() / group_count;
+        const WARP_SIZE: usize = 32;
+        let block_size = WARP_SIZE;
+        let block_count = (input.len() + block_size) / block_size;
```

Comment:
This should make it so we don't have an additional empty block/threadgroup when `input` is divisible by `block_size`:
```suggestion
        let block_count = (input.len() + block_size - 1) / block_size;
```

Reply:
you are right! fixed